### PR TITLE
some fixes to aiturret code

### PR DESCRIPTION
**Pull request by Goober5000**
I did a careful review of all the functions in aiturret.cpp.  This cleans up the code a bit to more closely follow the original code flow from the public code release (for ease of comparison) and fixes several bugs.

1) fixed other weapons on a turret being handicapped by spawning weapon fire rate

2) fixed beam-free flag on a turret with multiple weapons

3) fixed turret accuracy penalty never being applied

4) fixed weapon timestamp being overridden

5) fixed player sound not being played, if specified, with swarm turret weapons

6) fixed flak not jittering if it had the fire-down-normals flag

This finally resolves Mantis 804.
http://scp.indiegames.us/mantis/view.php?id=804

### CODE REVIEW

File: code/ai/aiturret.cpp [hunk 1]
```diff
@@ -2380,105 +2361,119 @@ void ai_fire_from_turret(ship *shipp, ship_subsys *ss, int parent_objnum)
...
+		// Similar check for small weapons
+		if ( wip->wi_flags[Weapon::Info_Flags::Small_only] ) {
+			if ( lep->type != OBJ_SHIP) {
+				tentative_return = true;
+				continue;
+			}
+			if ( Ship_info[Ships[lep->instance].ship_info_index].is_big_or_huge() ) {
+				tentative_return = true;
+				continue;
+			}
```
Comment (Baezon) [inline code/ai/aiturret.cpp:2401]: This changes the logic of `small_only` targeting. In the original, `small_only` weapons _can_ be fired at non-ships, basically the inverse of huge (must be a ship AND big|huge).

**Review by Baezon — CHANGES_REQUESTED** [review]
Looks pretty good aside from one small issue. I didn't quite realize how inconsistent `Use_multiple_guns` would have been before. And I forgot about the flak jittering aim with the `fire_down_normals` change, sorry!

File: code/ai/aiturret.cpp (hunk 1 above)
Comment (Goober5000) [reply code/ai/aiturret.cpp:2397]: Ah, good catch.  New commit pushed.

Comment (Goober5000) [reply code/ai/aiturret.cpp:2397]: And no worries about missing the `fire_down_normals` check.  We all miss something occasionally!

**Review by Baezon — APPROVED** [review]

**Pull request merged**